File: 2_optional_class_member/main.cpp
#include <iostream>
#include <string>
#include <optional>

class UserProfile {
private:
	std::string username;
	std::optional<std::string> biography;

public:
	UserProfile(const std::string& username, const std::optional<std::string>& biography = std::nullopt)
		: username(username), biography(biography) {}

	void setBiography(const std::string& bio){
		biography = bio;
	}

	void clearBiography() {
		biography = std::nullopt;
	}

	void printProfile() const {
		std::cout << "Username: " << username << std::endl;
		if (biography) {
			std::cout << "Biography: " << *biography << std::endl;
		} else {
			std::cout << "No biography." << std::endl;
		}
	}
};

int main() {
	UserProfile user("alice");
	user.printProfile();
	user.setBiography("Enthusiast programmer and tech lover!");
	user.printProfile();
	return 0;
}

File: 2_optional_class_member/main.py
class UserProfile:
	def __init__(self, username, biography=None):
		self.username = username
		self.biography = biography

	def set_biography(self, bio):
		self.biography = bio
	
	def clear_biography(self):
		self.biography = None

	def print_profile(self):
		print("Username:", self.username)
		if self.biography is not None:
			print("Biography: ", self.biography)
		else:
			print("No biography")

user = UserProfile("alice")
user.print_profile()
user.set_biography("A tech lover.")
user.print_profile()
